src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,41 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> dict:
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
+        spike_vector["unit_index"].astype(np.int64),
+        spike_vector["segment_index"].astype(np.int64),
+    )
+
+    return spike_trains
+
+
+if HAVE_NUMBA:
+
+    @numba.jit((numba.int64[::1], numba.int64[::1], numba.int64[::1]), nopython=True, nogil=True, cache=True)
+    def _vector_to_dict(sample_index, unit_index, segment_index):
```

Comment:
Needs a docstring as well,

Reply:
I though Samuel said it was unnecessary for stuff the user doesn't interact with (such as ExtractorSegments or functions starting with `_`)?

Reply:
If that is what @samuelgarcia believes I think he is wrong on that one. 

Documentation is also for other developers. Also for your future self.

For this specific one it makes my reviewing slower and leads to misundersandings. I don't know what this function is doing  by the name or what is the intention for example. There is a lot of implicit knowledge and intent on it that could be solved with some comments.

Reply:
Alright, I'll make one then!